### PR TITLE
Change OTLP HTTP content_type default to binary

### LINKED ISSUE #2485: [EXPORTER] Change OTLP HTTP content_type default to binary
The OTLP HTTP exporter uses JSON by default.

According to the spec, the export format by default should be BINARY instead.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specify-protocol

> If no configuration is provided the default transport SHOULD be http/protobuf unless SDKs have good reasons to choose grpc as the default (e.g. for backward compatibility reasons when grpc was already the default in a stable SDK release).

Issue found by kkuehl in PR #2480

### PR DESCRIPTION
**Pull request by owent**
Fixes #2485 

## Changes

+ Change OTLP HTTP content_type default to binary

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by marcalff — APPROVED** [review]
LGTM.

Thanks for the full fix, including environment variables.

File: exporters/otlp/src/otlp_http.cc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+HttpRequestContentType GetOtlpHttpProtocolFromString(nostd::string_view name) noexcept
+{
+  if (name == "http/json")
+  {
+    return HttpRequestContentType::kJson;
+  }
+  else
+  {
+    return HttpRequestContentType::kBinary;
+  }
```
Comment (esigo) [inline exporters/otlp/src/otlp_http.cc:21]: nit
```suggestion
  return HttpRequestContentType::kBinary;
```

**Review by esigo — APPROVED** [review]
LGTM
thank for the PR

**Review by lalitb — APPROVED** [review]
LGTM. Thanks for the changes.

**Pull request merged**